### PR TITLE
Add critical section to the card component

### LINKED ISSUE #9882: Add critical background and border to the Card component
## Issue summary
We needed to introduce a new card to the ProductDetails page, tailored specifically for suspended products. This card will display relevant information when a product is in the suspended state and provide the user with options to delete the product or directly submit a counter-notice. In order to do that we need to have the option to have background and border with critical color in the Polaris Card component. 

Screenshot 2023-08-03 at 11 51 23 AM

## Steps to reproduce the problem

1. First step
1. Second step

## Reduced test case

The best way to get your bug fixed is to provide a reduced test case. This [CodeSandbox template](https://codesandbox.io/s/q82mlq0m26) is a great starting point.

## Specifications

- Are you using the React components? (Y/N):
- Polaris version number:
- Browser:
- Device:
- Operating System:

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash

```

### PR DESCRIPTION
**Pull request by fatimasajadi**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/9882

### WHAT is this pull request doing?

We needed to introduce a new card to the ProductDetails page, tailored specifically for suspended products. This card will display relevant information when a product is in the suspended state and provide the user with options to delete the product or directly submit a counter-notice. In order to do that we need to have the option to have background and border with critical color in the Polaris Card component. This PR adds the necessary styles in order to achieve that ability.

[Storybook Link](http://192.168.2.18:6006/?path=/story/all-components-card--with-critical-section)
Screenshot 2023-08-03 at 2 21 00 PM

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Card/Card.stories.tsx [hunk 1]
```diff
@@ -117,3 +117,27 @@ export function WithSubduedSection() {
     </Card>
   );
 }
+
+export function WithCriticalSection() {
+  const {polarisSummerEditions2023} = useFeatures();
```
Comment (kyledurand) [inline polaris-react/src/components/Card/Card.stories.tsx:122]: Did you mean to use this somewhere?

File: polaris-react/src/components/Card/Card.tsx [hunk 2]
```diff
@@ -30,13 +30,15 @@ export interface CardProps {
   padding?: Spacing;
   /** Border radius value above a set breakpoint */
   roundedAbove?: BreakpointsAlias;
+  hasCriticalStatus?: boolean;
```
Comment (kyledurand) [inline polaris-react/src/components/Card/Card.tsx:33]: We're moving away from boolean props like this. It should be `tone` with `critical` option

File: polaris-react/src/components/Card/Card.tsx [hunk 3]
```diff
@@ -87,6 +89,10 @@ export const Card = ({
           borderRadius={hasBorderRadius ? defaultBorderRadius : undefined}
           overflowX="hidden"
           overflowY="hidden"
+          borderColor={
+            hasCriticalStatus ? 'border-critical-subdued' : undefined
+          }
+          borderWidth="1"
```
Comment (kyledurand) [inline polaris-react/src/components/Card/Card.tsx:95]: This is the non-uplifted box so you wouldn't see this in the admin

**Review by kyledurand — CHANGES_REQUESTED** [review]
I'd like to focus on what's blocking this card from being built in web vs in Polaris. The only thing I see is that we don't export the ShadowBevel component which is used to build the uplifted card.

The other challenge there is getting the critical border to show up with the ShadowBevel wrapper on Box.

cc @aaronccasanova. I do understand the need for this component but I think we should explore other options before baking it into Polaris

**Comment by kyledurand** [comment]
[This is what I mean by composability](https://codesandbox.io/s/priceless-torvalds-rx3nty?file=/App.tsx) I don't think this needs to be a part of Polaris but let me know if you run into anything Polaris is missing, any blockers, and I'll give you a hand

**Comment by fatimasajadi** [comment]
> [This is what I mean by composability](https://codesandbox.io/s/priceless-torvalds-rx3nty?file=/App.tsx) I don't think this needs to be a part of Polaris but let me know if you run into anything Polaris is missing, any blockers, and I'll give you a hand

I agree with you. We should not need to change anything in Polaris, in order to achieve what we wanted. Thanks so much for taking the time and explain this. I will share this with my team and see if there is anything I'm missing.

**Comment by fatimasajadi** [comment]
We all agreed that we do not need to make changes to root Polaris in order to build the custom card for suspended products.

**Pull request closed without merging**